packages/web/src/components/test-tracks-table/TestTracksTable.tsx

```diff
@@ -0,0 +1,483 @@
+                e.stopPropagation()
+                onClickFavorite?.(track)
+              }}
+              favorited={track.has_current_user_saved}
+            />
+          </div>
+        </Tooltip>
+      )
+    },
+    [onClickFavorite, userId]
+  )
+
+  const renderRepostButtonCell = useCallback(
+    (cellInfo) => {
+      const track = cellInfo.row.original
+      const deleted = track.is_delete || track.user?.is_deactivated
+      if (deleted) return null
+      const isOwner = track.owner_id === userId
+      return isOwner ? null : (
+        <Tooltip text={track.has_current_user_reposted ? 'Unrepost' : 'Repost'}>
+          <div>
+            <TableRepostButton
+              className={cn(styles.tableActionButton, {
+                [styles.active]: track.has_current_user_reposted
+              })}
+              onClick={(e) => {
+                e.stopPropagation()
+                onClickRepost?.(track)
+              }}
+              reposted={track.has_current_user_reposted}
+            />
+          </div>
+        </Tooltip>
+      )
+    },
+    [onClickRepost, userId]
+  )
+
+  const renderOverflowMenuCell = useCallback(
+    (cellInfo) => {
+      const track = cellInfo.row.original
+      const deleted = track.is_delete || !!track.user.is_deactivated
+      return (
+        <OverflowMenuButton
+          className={styles.tableActionButton}
+          onClick={(e) => {
+            e.stopPropagation()
+          }}
+          isDeleted={deleted}
+          // onRemove={props.onClickRemove}
+          // removeText={props.removeText}
+          handle={track.handle}
+          trackId={track.track_id}
+          uid={track.uid}
+          date={track.date}
+          isFavorited={track.has_current_user_saved}
+          isOwner={track.owner_id === userId}
+          isOwnerDeactivated={!!track.user.is_deactivated}
+          isArtistPick={track.user._artist_pick === track.track_id}
+          index={cellInfo.row.index}
+          trackTitle={track.name}
+          albumId={null}
+          albumName={null}
+          trackPermalink={track.permalink}
+        />
+      )
+    },
+    [userId]
+  )
+
+  const renderTrackActions = useCallback(
+    (cellInfo) => {
+      return (
+        <div className={styles.trackActionsContainer}>
+          {renderRepostButtonCell(cellInfo)}
+          {renderFavoriteButtonCell(cellInfo)}
+          {renderOverflowMenuCell(cellInfo)}
+        </div>
+      )
+    },
+    [renderFavoriteButtonCell, renderOverflowMenuCell, renderRepostButtonCell]
+  )
+
+  // Columns
+  const tableColumnMap: Record<
+    TracksTableColumn,
+    Partial<ColumnInstance>
+  > = useMemo(
+    () => ({
+      artistName: {
+        id: 'artistName',
+        Header: 'Artist',
+        accessor: 'artist',
+        Cell: renderArtistNameCell,
+        maxWidth: 300,
+        width: 120,
+        sortType: ({ original: rowA }: any, { original: rowB }: any) =>
+          alphaSorter(rowA, rowB, 'artist'),
```

Comment:
Not super important but you could curry the sorter functions to make these calls cleaner, for example if you change the signature of alphaSorter to

```ts
const alphaSorter = (key) => (rowA, rowB) => {}
```

it would allow you to do this in the col defs:
```ts
sortType: alphaSorter('artist')
```

Reply:
🧠